File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_351_A.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main() {
    int n, i, o;
    float arr_sum, res;
    scanf("%d", &n);
    float arr[n];

    for(i = 0; i < n; i++) {
        scanf("%f", &arr[i]);
        arr[i] = arr[i] - (int)arr[i];
    }

    float temp;
    for(i = 0; i < n; i++) {
        for(int j = i+1; j < n; j++) {
            if(arr[j] < arr[i]) {
                temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
    }

    o = 2 * n - n;
    arr_sum = 0;
    for(i = 0; i < n; i++) {
        arr_sum += arr[i];
    }

    res = 2000000000;
    for(i = 0; i < n + 1; i++) {
        if(i + o >= n) {
            res = fmin(res, fabs(i - arr_sum));
        }
    }

    printf("%.3f\n", res);

    return 0;
}

// End of Code

File: LLM-Translations/GPT-4/avatar/Python/Java/codeforces_351_A.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class codeforces_351_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();
        List<Double> arr = new ArrayList<>();
        for (String s : sc.nextLine().split(" ")) {
            arr.add(Double.parseDouble(s));
        }
        arr.sort((x, y) -> Double.compare(x - (int) x.doubleValue(), y - (int) y.doubleValue()));
        arr.removeIf(x -> x - (int) x.doubleValue() == 0);
        int o = 2 * n - arr.size();
        double arrSum = arr.stream().mapToDouble(x -> x - (int) x.doubleValue()).sum();
        double res = (int) 2e9;
        for (int i = 0; i < n + 1; i++) {
            if (i + o >= n) {
                res = Math.min(res, Math.abs(i - arrSum));
            }
        }
        System.out.printf("%.3f%n", res);
    }
}
// End of Code
